File: Antlr/src/main/scala/io/github/chaosunity/casc/parsing/scope/LocalVariable.scala
package io.github.chaosunity.casc.parsing.scope

import io.github.chaosunity.casc.parsing.`type`.Type
import io.github.chaosunity.casc.parsing.expression.Expression

class LocalVariable(_type: Type, val name: String) extends Expression {
    override def `type`: Type = _type

    override def negative: Boolean = false
}

File: src/main/kotlin/io/github/chaosunity/casc/parsing/scope/LocalVariable.kt
package io.github.chaosunity.casc.parsing.scope

import io.github.chaosunity.casc.parsing.Immutable
import io.github.chaosunity.casc.parsing.type.Type
import io.github.chaosunity.casc.parsing.type.Variable

data class LocalVariable(override val name: String, override var type: Type, override val immutable: Boolean = true) : Variable, Immutable